src/components/app.tsx

```diff
@@ -49,28 +53,51 @@ function App() {
   };
 
   const getData = async () => {
-    setStatusMessage("Fetching data...");
+    setStatusMessage(strings.fetchingMsg);
     setStatusGraphic(<ProgressIndicator/>);
     const res = await createTableFromSelections(selectedOptions);
     if (res !== "success") {
-      setStatusMessage("Fetch Error. Please retry.");
+      setStatusMessage(strings.fetchSuccess);
       setStatusGraphic(<Error/>);
     } else {
-      setStatusMessage("Fetched data.");
+      setStatusMessage(strings.fetchError);
       setStatusGraphic(<Checkmark/>);
     }
   };
 
   const handleGetData = async () => {
     const numberOfRows = getNumberOfItems(selectedOptions);
+
+    const attrKeys = attributeOptions.filter((attr) => attr.key !== "cropUnits").map((attr) => attr.key);
+    const selectedAttrKeys = attrKeys.filter((key) => selectedOptions[key].length > 0);
+    const allSelectedAttrs = flatten(selectedAttrKeys.map((key) => selectedOptions[key]));
+    const selectedYears = selectedOptions.years;
+
+    allSelectedAttrs.map((attr) => {
+      const attrInfo = queryData.find((q) => q.plugInAttribute === attr);
+      if (attrInfo) {
+        const availableYears = attrInfo.years[selectedOptions.geographicLevel];
+        for (let i = 0; i < selectedYears.length; i ++) {
+          const y = selectedYears[i];
+          if (!availableYears.includes(y)) {
+            setWarningMessage(strings.yearsWarning);
+            setShowWarning(true);
+            break;
+          }
+        }
+      }
+    });
+
```

Comment:
Do you want to exit early here if there is a warning set on line 83?  Otherwise if might be overwritten on line 92.

Reply:
Thanks Doug, I slightly changed up the approach in case both warnings are needed at the same time and I've re-requested your review :)